Gems/Atom/RPI/Code/Include/Atom/RPI.Public/SceneBus.h

```diff
@@ -39,14 +39,37 @@ namespace AZ
             template<typename Bus>
             using ConnectionPolicy = SceneConnectionPolicy<Bus>;
             //////////////////////////////////////////////////////////////////////////
+
+            enum class RenderPipelineChangeType
+            {
+                Added,              //The render pipeline was added to this scene
+                PassChanged,      //Any passes of this render pipeline are modified before a render tick
+                Removed             //The render pipeline was removed from this scene
+            };
                         
+            //! O3DE_DEPRECATION_NOTICE(GHI-12687)
+            //! @deprecated use OnRenderPipelineChanged(RenderPipeline*, RenderPipelineChangeType::Added)
             //! Notifies when a render pipeline is added to this scene. 
             //! @param pipeline The render pipeline which was added
             virtual void OnRenderPipelineAdded(RenderPipelinePtr pipeline) {};
+                        
+            //! O3DE_DEPRECATION_NOTICE(GHI-12687)
+            //! @deprecated use OnRenderPipelineChanged(RenderPipeline*, RenderPipelineChangeType::PassChanged)
+            //! Notifies when any passes of this render pipeline are modified before a render tick
+            //! This includes adding a pass, removing a pass, or if pass data changed (such as attachments, draw list tags, etc.)
+            //! Feature processors may need to use it to update their cached pipeline states
+            //! @param pipeline The render pipeline which was modified
+            virtual void OnRenderPipelinePassesChanged([[maybe_unused]] RenderPipeline* renderPipeline) {}
 
+            //! O3DE_DEPRECATION_NOTICE(GHI-12687)
+            //! @deprecated use OnRenderPipelineChanged(RenderPipeline*, RenderPipelineChangeType::Removed)
             //! Notifies when a render pipeline is removed from this scene.
             //! @param pipeline The render pipeline which was removed
             virtual void OnRenderPipelineRemoved([[maybe_unused]] RenderPipeline* pipeline) {};
+
+            //! Notifies when a render pipeline was added, removed or changed
+            //! @param pipeline The render pipeline which was added
+            virtual void OnRenderPipelineChanged(RenderPipeline* , RenderPipelineChangeType) {};
```

Comment:
There is RenderPipelinePtr only used for adding behavior in the original API. But why enventually RenderPipeline* is used over RenderPipelinePtr?

Reply:
We don't want user to keep a smart pointer of the render pipeline especially when it's about to be removed.